docs/rfcs/0007-deployment-chain.md

```diff
@@ -0,0 +1,129 @@
+
+![image](assets/deployment-chain-region.png)
+
+All the above requirements share the same thing in the context, that is: it can be done by users' deployment for their applications as PipeCD applications one by one and make trigger those deployments manually via console or make trigger via pull requests to those application configurations separately, but this all manual stub are tedious and difficult to manage smoothly. With this new __PipeCD deployment chain__ feature, all of those manual steps will be replaced, to keep the good point of using a CD system.
+
+# Detailed design
+The idea is to keep the PipeCD application deployment as a unit of deployment as is, but add __a way to enable users to manipulate a deployment chain based on the state of the last sucessfully run deployment__.
+
+A canonical flow would look as below:
+- Users trigger to run their first application (the first application in their deployment chain) via the web console or pull requests as usual (1)
+- Deploy the first triggered application in the deployment chain (2)
+- After the application is deployed successfully, the piped will send a trigger command to trigger the deployment chain to the control-plane, since all those applications are shared in the same project, this trigger should be valid (3)
+- Control-plane gets deployment events triggered by the first application's piped and makes sync deployment commands for all listed applications (4)
+- Pipeds which handle deployment for all triggered applications fetch sync command and deploy its application once the conditions are satisfied (5)
+
+![image](assets/deployment-chain-idea.png)
+
+## Configuration
+
+At the step (3), to enable configure the deployment chain after the first applications (in the deployment chain) deployed successfully, we introduce a new configuration section to the deployment configuration named `postSync` which contains configurations of actions that the in charge piped should do after the deployment run successfully.
+
+The configuration would look like:
+
+```yaml
+apiVersion: pipecd.dev/v1beta1
+kind: Terraform
+spec:
+  input:
+    ...
+  pipeline:
+    ...
+  trigger:
+    onOutOfSync:
+      disable: true # default is `true`
+    onCommand:
+      disable: false # default is `false`
+    onCommit:
+      disable: false # default is `false`
+      paths:
+        - manifests/deployment.yaml
+    onChain:
+      disable: false # default is `false`
+  postSync:
+    chain:
+      applications:
+        - name: app2
+          labels:
+            - app: 2 # will match all app2-1, app2-2, app2-3 which share label (app: 2)
+        - name: app3
+          kind: Kubernetes
+          labels:
+            - app: 3 # will match all app which share label (app: 3)
+      conditions:
+        - commitPrefix: “xx”
+        - ...
```

Comment:
It could be great to have this kind of example for each scenario we described above.
- promoting from dev to stag to prod
- rolling out to multiple regions

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2743/commits/58cfaf403390d76d1db495bb219c0ad7b80f8555